runtime/interpreter/value.go

```diff
@@ -11541,14 +11541,11 @@ func NewPublicKeyValue(
 		sema.PublicKeyVerifyPoPFunction: publicKeyVerifyPoPFunction,
 	}
 
-	// Validate the public key, and initialize 'isValid' field.
-
-	publicKeyValue.SetMember(
-		interpreter,
-		getLocationRange,
-		sema.PublicKeyIsValidField,
-		validatePublicKey(interpreter, getLocationRange, publicKeyValue),
-	)
+	// Validate the public key.
+	isValid := validatePublicKey(interpreter, getLocationRange, publicKeyValue)
```

Comment:
In the call to `validatePublicKey`, there will be a call to `runtimeInterface.ValidatePublicKey` which errors if the key is invalid. In that case there will be an internal panic with an error different than `InvalidPublicKeyError`. 
It would be good to return that internal panic as `InvalidPublicKeyError` too for consistency.

Moreover, that internal error contains the reason for the key invalidity, and it would be good to log it to the user. Maybe `func (e *InvalidPublicKeyError) Error()` should log it ?

Reply:
Can you point me to an example of logging we do at runtime? Or do you mean including the original error in the InvalidPublicKeyError struct, and including it in `fmt.Sprintf`?

Reply:
I meant the latter, including the original error text in the returned `InvalidPublicKeyError`